\section{Application to neuronal differentiation}\label{sec:neuroseq}

Next, we considered single-cell transcriptomic data from over 200 individuals from \cite{jerber2020population}.
We focused on a single cell type: midbrain dopaminergic neurons (DA), across three conditions defined in the original publication: day 30, day 52 untreated, and day 52 rotenone-treated. 
In total, this consists of 135,435 cells across 210 donors.


\subsection{Preprocessing of scRNA-seq count data}
Count data were processed as in the primary paper \cite{jerber2020population}, where counts were normalised using scanpy \cite{wolf2018scanpy} and log-transformed ($log_2(x+1)$).
Prior to be inputed into the model as phenotype vectors (i.e., as $\mathbf{y}$), single-cell counts for a given gene were quantile-normalized to better fit the Gaussian distribution assumed by the model.


\subsection{Pseudo-cell calculation}

We obtained pseudo-cells by clustering similar cells in UMAP space using an approach similar to approaches described in \cite{baran2019metacell, detomaso2019functional}. 
Briefly, for each cell type we calculated the first 50 PCs across all conditions and donors, using all expressed genes (after QC, n=32,738). 
Next, we applied batch-correction for the pool id using Harmony \cite{korsunsky2019fast}, as implemented in scanpy (`scanpy.external.harmony\_integrate'). 
The Harmony-corrected PCs were then used to build a k-NN (k=10) graph based on Euclidean distances, separately for each condition and donor. 
Subsequently, cells were clustered using scanpyâ€™s implementation of the Leiden algorithm (\cite{traag2019louvain}, as implemented in `scanpy.tl.leiden') at a resolution of 3.4. \\

The pseudo-cell calculation described above resulted, in these data, in a total of 8,479 pseudocells (10-40 pseudocells per donor, 10-100 cells per pseudocell).

\subsection{Clustering of single-cell effect sizes}

Next, we set out to cluster context-specific eQTL based on their allelic effects due to G$\times$C.
First, we considered 212 eQTL which displayed significant G$\times$C effects using our method (FDR $<$ 5\%) when considering the leading 10 MOFA factors as cellular contexts.
Next, we considered the estimated effect size profiles due to G$\times$C (as in Section 2).
These profiles were normalised to contain only positive values between 0 and 1.
Finally, clustering on the normalised profiles was performed using SpatialDE \cite{svensson2018spatialde}, by using the same first 10 MOFA factors as spatial coordinates, and default parameters. 
This identified 12 clusters, containing between 2 and 43 genes (main text Fig. 4b).

\subsection{Cluster enrichment}

For each cluster, we considered Pearson's correlation between the cluster's summary profiles (as outputted by SpatialDE) and single-cell gene expression across all genes (n=32,738).
Next, we selected all genes with positive correlation larger than 0.4 and used gprofiler \cite{raudvere2019g} to identify enriched pathways (considering Gene Ontology (GO) biological processes, molecular function, cellular components, pathways from KEGG Reactome and WikiPathways; miRNA targets from miRTarBase and regulatory motif matches from TRANSFAC; tissue specificity from Human Protein Atlas; protein complexes from CORUM and human disease phenotypes from Human Phenotype Ontology). 
The gprofiler function ("g:GOSt" as implemented in R) performs over-representation analysis on input gene list (ordered by correlation level) using a hypergeometric test, corrected for multiple testing.
The latter is performed using a tailor-made method (g:SCS algorithm) which analytically approximates a threshold t corresponding to the 5\% upper quantile of randomly generated queries of the provided size. 
All actual p-values resulting from the query are transformed to corrected p-values by multiplying these to the ratio of the approximate threshold t and the initial experiment-wide threshold a=0.05.).
Finally, we selected the top 3 significant (adjusted p-values $<$ 0.05) pathways per cluster (Supplementary Fig. 4.2). 