\section{Predicting cell-specific effect sizes driven by GxC interactions}
\label{sec:beta_gxe}

Using CellRegMap it is possible to estimate cell-level allelic effects due to GxC, (thus estimating $\boldsymbol{\beta_{GxC}}$ from eq. \eqref{eq:scStructLMM_marginal}) for each gene-SNP pair tested.
For this derivation, we will use the function representation of linear mixed model as a Gaussian process ($\mathcal{GP}$).
Briefly (for more details see section \ref{sec:derivations_blup}), 

% \begin{equation}\label{eq:GP}
%     \mathbf{f}(\mathbf{X}) \sim \mathrm{GP}(\mathbf{m}(\mathbf{x}), k(\mathbf{x},\mathbf{x}^T)),
% \end{equation}

\begin{align}
    \mathbf{f} \sim \mathcal{GP}\{\mathbf{m}; k(\mathbf{X}, \mathbf{X})_{\bftheta}\},
\end{align}

or:

\begin{align}
    \mathbf{f} = \mathbf{m} + \mathbf{c},
\end{align}

where we define:

\begin{align}
    \mathbf{m} = \mathbf{X} \bfbeta_{\theta}
     \text{~~and~~}
     \mathbf{c} \sim \mathcal N(\mathbf{0}, k(\mathbf{X}, \mathbf{X})_{\theta}). 
\end{align}

When we model our data ($\mathbf{X}$ and $\mathbf{y}$), we consider $\mathbf{y}$ as being a sample from the random variable $\mathbf{f}$, and $\mathbf{X}$ as being fixed.
Let us collectively call the parameters $\boldsymbol{\theta}$ and  $\hat{\boldsymbol{\theta}}_{BLUP}$ (or $\hat{\boldsymbol{\theta}}$) their best linear unbiased predictor (BLUP) estimator \cite{robinson1991blup}. \\

For out of sample ($*$) prediction, we can define \cite{rasmussen2003gaussian}:

\begin{align}
    \mathbf{y}_* \coloneqq \mathrm{E}[\mathbf{f}_* | \mathbf{y}]_{\hat{\boldsymbol{\theta}}},
\end{align}

which can be written as (see section \ref{sec:derivations_blup} for intermediate steps):

\begin{align}\label{eq:y*}
\mathbf{y}_* = \mathbf{m}_* + k(\mathbf{X}_*,\mathbf{X})k(\mathbf{X},\mathbf{X})^{-1}(\mathbf{y}-\mathbf{m}).
\end{align}

% the best estimator of the out-of-sample prediction for $\mathbf{f}_*$  is its best linear unbiased predictor (BLUP), defined as its expected value condition on $\mathbf{f}$ and $\mathbf{X},\mathbf{X}_*$\cite{rasmussen2003gaussian}:

% \begin{equation}
%     E[\mathbf{f}_*|\mathbf{f}] = \mathbf{m}_* +k(\mathbf{X}_*,\mathbf{X})k(\mathbf{X},\mathbf{X})^{-1}(\mathbf{f}-\mathbf{m}).
% \end{equation}

In our case, we use a modified model from eq. \eqref{eq:scStructLMM} where the additive environments are modelled as fixed effects: 

\begin{equation}\label{eq:scStructLMM_E_as_FE}
    \mathbf{y} = \overbrace{\mathbf{W}\boldsymbol{\alpha} + \mathbf{g}\beta_G + \mathbf{E}\boldsymbol{\gamma}}^{\mathbf{m}} + \overbrace{\mathbf{g} \odot \boldsymbol{\beta}_{GxE}+ \mathbf{u} + \boldsymbol{\psi}}^{\mathbf{c}}.
\end{equation}

% \begin{equation}
%     \mathbf{f}(\mathbf{X}) = \mathbf{y} = \mathbf{W}\boldsymbol{\alpha}+\mathbf{g}\beta_G+\mathbf{g} \odot \boldsymbol{\beta}_{GxE}+\mathbf{e} + \mathbf{u} + \boldsymbol{\psi},
% \end{equation}

By substituting $\mathbf{m}$ and $\mathbf{c}$ into eq. \eqref{eq:y*} we obtain a formula for $\mathbf{y}_*$:

\begin{equation}\label{eq:y*_expanded}
    \mathbf{y}_* = \mathbf{W}_*\bfalpha + \bfg_* \beta_G + \mathbf{E}_* \boldsymbol{\gamma} + k(\mathbf{X},\mathbf{X}_*) k(\mathbf{X},\mathbf{X})^{-1} (\mathbf{y}-\mathbf{W}\bfalpha + \bfg \beta_G + \mathbf{E} \boldsymbol{\gamma}).
\end{equation}

% with ($\mathbf{X} = {\mathbf{W},\mathbf{g},\mathbf{E},\mathbf{R}}$) such that:

% \begin{equation}
%     \mathbf{m}_*(\mathbf{X}) = \mathbf{W}_*\boldsymbol{\alpha}_{*} + \mathbf{g}_*\beta_G + \mathbf{E}_*\boldsymbol{\gamma}
% \end{equation}

% \begin{equation}
%     k(\mathbf{X},\mathbf{X}) = \sigma_{GxE}^2(\mathbf{g}\odot\mathbf{E})(\mathbf{g}\odot\mathbf{E})^T
% \end{equation}

% and 

% \begin{equation}
% \mathbf{y}_{*}^{BLUP} = E[\mathbf{y}_*|\mathbf{y}] = \mathbf{W}_*\boldsymbol{\alpha}_{*}+\mathbf{g}_*\beta_G+\mathbf{E}_*\gamma + k(\mathbf{X}_*\mathbf{X})\mathbf{K}^{-1}(\mathbf{y}-\mathbf{W}\boldsymbol{\alpha}-\mathbf{g}\beta_G-\mathbf{E}\gamma).
% \end{equation}

Now, to obtain an estimate for $\boldsymbol{\beta}_*$, we can evaluate eq. \eqref{eq:y*_expanded} when $\mathbf{g}_*=0$ ($\mathbf{y}_*(\mathrm{ref}$)) and when $\mathbf{g}_*=1$ ($\mathbf{y}_*(\mathrm{alt}$)), and consider the difference, scaled by a number based on the MAF\footnote{minor allele frequency} ($p$):

\begin{equation}
    \bfbeta_{GxC}^* = \frac{1}{\sqrt{2p(1-p)}} \ (\mathbf{y}_*(\mathrm{alt}) - \mathbf{y}_*(\mathrm{ref})),
\end{equation}

from which, skipping a few steps:

\begin{align}
    \bfbeta_{GxC}^* = \frac{1}{\sqrt{2p(1-p)}} \ \{\mathbf{1}\beta_G + \sigma_{GxC}^2\mathbf{E}_*(\mathbf{g}\odot\mathbf{E})^T \mathbf{K}^{-1}(\mathbf{y}-\mathbf{W}\bfalpha - \bfg \beta_G - \mathbf{E} \boldsymbol{\gamma})\},
\end{align}

where $\mathbf{K} = \mathrm{Cov}(\mathbf{y}) = \sigma_{GxE}^2(\mathbf{g}\odot\mathbf{E})(\mathbf{g}\odot\mathbf{E})^T + \sigma_{re}^2 \mathbf{R} \odot \mathbf{E}\mathbf{E}^T + \sigma_n^2 \mathbf{I}$. \\

By setting $\mathbf{E}_* = \mathbf{E}$, we can perform in-sample estimation of allelic effects.


% \newpage

% \subsection{Aggregate environment driving GxE}

% Using a similar approach, we can also identify ...
% We consider the same modified model as above (eq. \eqref{eq:scStructLMM_E_as_FE}):

% \begin{equation}
%     \mathbf{y} = \mathbf{W}\boldsymbol{\alpha} + \mathbf{g}\beta_G + \mathbf{E}\boldsymbol{\gamma} + \mathbf{g} \odot \boldsymbol{\beta}_{GxE}+ \mathbf{u} + \boldsymbol{\psi},
% \end{equation}

% such that: 

% \begin{equation}
%     Cov(\mathbf{y}) = \sigma^2_1 \{\rho_1 (\mathbf{g} \odot \mathbf{E})(\mathbf{g} \odot \mathbf{E})^T + (1-\rho_1)\mathbf{R}\odot\boldsymbol{\Sigma}\} + \sigma_n^2 \mathbf{I}.
% \end{equation}

% The coefficient can be written as:

% \begin{equation}
%     \boldsymbol{\delta} = \sigma_{GxE}^2 \ \mathbf{g} \odot \mathbf{E} \ Cov(\mathbf{y})^{-1}(\mathbf{y}-\mathbf{W}\boldsymbol{\alpha}-\mathbf{g}\beta_G-\mathbf{E}\gamma),
% \end{equation}

% and consequently the aggregate environment is:

% \begin{align}
%     \mathrm{Aggr. Env.} = \mathbf{E}\boldsymbol{\delta}.
% \end{align}

% % beta = sigma2 * gE.T @ Sigma-1 y\_adj

% % aggr env = E * $\beta$